### PR TITLE
Update bdalg.py

### PR DESCRIPTION
**Pull request by Marxlp**
Extend the series and parallel to support more parameters

### CODE REVIEW

File: control/bdalg.py [hunk 1]
```diff
@@ -60,14 +60,16 @@
 
 __all__ = ['series', 'parallel', 'negate', 'feedback', 'append', 'connect']
 
-def series(sys1, sys2):
-    """Return the series connection sys2 * sys1 for --> sys1 --> sys2 -->.
+def series(sys1, sys2,*sysn):
+    """Return the series connection (... * sys3 *) sys2 * sys1
```
Comment (roryyorke) [inline control/bdalg.py]: How about
```
def series(sys1, *sysn):
    from functools import reduce
    return reduce(lambda x, y: y*x, sysn, sys1)
```
and similarly for parallel?

One downfall there is that `series(s1)` would return the `s1` instance, and not a copy of it.

Comment (Marxlp) [inline control/bdalg.py]: Yes, it's  a good idea. But, using functional programming would increase the complexity of the code or decrease the elegance of the code.

**Comment by slivingston** [comment]
@Marxlp are you going to add unit tests?

**Comment by slivingston** [comment]
regarding code style, you must use 4-space indentation units.

File: control/bdalg.py (hunk 1 above)
Comment (slivingston) [reply control/bdalg.py]: Depending on one's experience, the use of `reduce` and `lambda` is elegant. Given the simplicity of the operation, the more important question here is performance. Sometimes functional idioms like the one suggested by @roryyorke can be better optimized by the Python compiler than a for-loop.

Comment (cwrowley) [reply control/bdalg.py]: +1 for `reduce` and `lambda`

**Comment by murrayrm** [comment]
Submitted [pull request](https://github.com/Marxlp/python-control/pull/1) to @Marxlp fixing the comments above.  Will wait a few days for update.  If not, will submit separate PR and merge to master.

**Comment by Marxlp** [comment]
I opened a new pull request.

**Comment by murrayrm** [comment]
For posterity: new pull request is #178.

**Pull request closed without merging**